runner.py

```diff
@@ -0,0 +1,647 @@
+import argparse
+import importlib.util
+import logging
+import os
+import pathlib
+import shutil
+import signal
+import subprocess
+import sys
+
+import yaml
+import serial
+import pexpect
+import pexpect.fdpexpect
+
+SCRIPT_PATH = pathlib.Path(__file__).absolute()
+PRTOS_TEST_DIR = SCRIPT_PATH.parents[0]
+PRTOS_PROJECT_DIR = SCRIPT_PATH.parents[1]
+_BOOT_DIR = PRTOS_PROJECT_DIR / "_boot"
+
+QEMU_CMD = {
+    'ia32-generic': [
+        ('qemu-system-i386',),
+        ('-hda', str(_BOOT_DIR) + '/phoenix-ia32-generic.disk'),
+        ('-nographic',),
+        ('-monitor', 'none'),
+        ('-serial', 'stdio')
+    ],
+    'riscv64-virt': [
+        ('qemu-system-riscv64',),
+        ('-bios', str(_BOOT_DIR) + '/phoenix-riscv64-virt.bbl'),
+        ('-machine', 'virt'),
+        ('-nographic',),
+        ('-monitor', 'none'),
+        ('-serial', 'stdio')
+    ]
+}
+
+TARGETS = ['ia32-generic']
+
+class Color:
+    FAIL = '\033[91m'
+    OK = '\033[92m'
+    IGNORE = '\033[93m'
+    END = '\033[0m'
+
+    @staticmethod
+    def colorify(string, color):
+        return color + string + Color.END
```

Comment:
```suggestion
        return f"{color}{string}{Color.END}"
```
It would be better  to use f-string.

Reply:
I agree it is more consistent.